framework/src/wifi_manager/wifi_manager.c

```diff
@@ -398,25 +514,50 @@ wifi_manager_result_e wifi_manager_set_mode(wifi_manager_mode_e mode, wifi_manag
 		result = wifi_utils_start_sta();
 		if (result != WIFI_UTILS_SUCCESS) {
 			ndbg("start STA fail (change STA mode fail)");
-			wifi_mutex_release(&w_mutex);
+			wifi_mutex_release(w_mutex);
 			return WIFI_MANAGER_FAIL;
 		}
 
 		g_manager_info.mode = STA_MODE;
-		g_manager_info.status = AP_DISCONNECTED;
-		wifi_mutex_release(&w_mutex);
+		wifi_status_set(AP_DISCONNECTED);
+		wifi_mutex_release(w_mutex);
 
-		nvdbg("Wifi Chnage SOFT AP MODE -> STA MODE");
+		nvdbg("Wifi mode is changed to station");
+	}
+	else {
+			ndbg("Invalid config: %d %d %s\n", mode, config->channel, config->ssid);
```

Comment:
In case that this code is executed, config is NULL exactly.
because you checked invalid mode as if (mode != STA_MODE && mode != SOFTAP_MODE) { return XX;}.
So this else case means STA_MODE mode with NULL config.

Reply:
I remove printing config->channel and config->ssid in ndbg().